src/include/sof/audio/module_adapter/intel/processing_module_adapter.h

```diff
@@ -0,0 +1,134 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ *
+ * Copyright(c) 2022 Intel Corporation. All rights reserved.
+ *
+ * Author: Jaroslaw Stelter <jaroslaw.stelter@intel.com>
+ *
+ */
+
+#ifndef _PROCESSING_MODULE_ADAPTER_H
+#define _PROCESSING_MODULE_ADAPTER_H
+
+#ifdef __cplusplus
+
+#include <processing_module_interface.h>
+#include <module_initial_settings.h>
+#include <adsp_stddef.h>
+#include <system_error.h>
+#include <adsp_stddef.h>
+#include <sof/audio/module_adapter/module/common.h>
+
+extern "C" {
+namespace dsp_fw
+{
+	/*!
+	 * \brief This ModuleAdapter class can adapt a ProcessingModuleInterface instance into
+	 *        a ModuleInstance instance.
+	 *
+	 * The overall base FW can only handle ModuleInstance object. Purpose of this adapter is
+	 * to turn an intel_adsp::ProcessingModuleInterface object into a ModuleInstance object.
+	 */
+	class ModuleAdapter
+	{
+	public:
+		ModuleAdapter(
```

Comment:
Is this a 1:1 mapping on the core C API of this class ? i.e. do all the methods and params now align ?